next_sparseconvnet/utils/train_utils.py

```diff
@@ -85,3 +86,33 @@ def valid_one_epoch_segmentation(net, criterion, loader, nclass = 3):
         print(loss_)
 
     return loss_epoch, iou_epoch
+
+
+def train_segmentation(nepoch, train_data_path, valid_data_path, train_batch_size, valid_batch_size, net, criterion, optimizer, checkpoint_dir, tensorboard_dir):
+    """
+        Trains the net nepoch times and saves the model anytime the validation loss decreases
+    """
+    train_gen = DataGen(train_data_path, LabelType.Segmentation, nevents = 100)
```

Comment:
nevents obviously shouldnt be fixed inside the function, you can add it as a keyword parameter with default value `None` so it takes all events